src/modules/Elsa.Workflows.Management/Contracts/IActivityRegistryPopulator.cs

```diff
@@ -27,4 +20,21 @@ public interface IActivityRegistryPopulator
     /// <param name="workflowDefinitionId">The ID of the workflow definition.</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     Task AddToRegistry(Type providerType, string workflowDefinitionId, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Removes workflow definition activities from the <see cref="IActivityRegistry"/>.
+    /// </summary>
+    /// <param name="providerType">The type of the Activity Provider.</param>
+    /// <param name="workflowDefinitionId">The ID of the workflow definition to remove.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    void RemoveDefinitionFromRegistry(Type providerType, string workflowDefinitionId, CancellationToken cancellationToken = default);
+    
+    
+    /// <summary>
+    /// Removes a workflow definition version activity from the <see cref="IActivityRegistry"/>.
+    /// </summary>
+    /// <param name="providerType">The type of the Activity Provider.</param>
+    /// <param name="workflowDefinitionVersionId">The ID of the workflow definition to remove.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    void RemoveDefinitionVersionFromRegistry(Type providerType, string workflowDefinitionVersionId, CancellationToken cancellationToken = default);
```

Comment:
The IActivityRegistry shouldn't be aware of workflow definitions; it should only be aware of activity descriptor providers.

I missed this the first time the first version of `AddToRegistry(..., workflowdefinitionId)` was added on April 10th, which is my bad. I think this is a mistake and should be revisited.

Reply:
Additionally, the "populater" exists only to populate the registry, so either we need to rename the service to e.g. "updater", or introduce a separate service called e.g. "depopulator" or "trimmer".